extcap_ot.py

```diff
@@ -0,0 +1,205 @@
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+import os
+import sys
+import tempfile
+import argparse
+import subprocess
+import threading
+
+from spinel.stream import StreamOpen
+from spinel.const import SPINEL
+from spinel.codec import WpanApi
+from serial.tools.list_ports import comports
+from enum import Enum
+
+# Nodeid is required to execute ot-ncp-ftd for its sim radio socket port.
+# This is maximum that works for MacOS.
+DEFAULT_NODEID = 34
+COMMON_BAUDRATE = [460800, 115200, 9600]
+
+class Config(Enum):
+    CHANNEL = 0
+    BAUDRATE = 1
+    TAP = 2
+
+def extcap_config(interface, option):
+    """List Configuration for the given interface"""
+    args = []
+    values = []
+    args.append((Config.CHANNEL.value, '--channel', 'Channel', 'IEEE 802.15.4 channel', 'selector', '{required=true}{default=11}'))
+    args.append((Config.TAP.value, '--tap', 'IEEE 802.15.4 TAP (only for Wireshark3.0 and later)', 'IEEE 802.15.4 TAP', 'boolflag', '{default=yes}'))
+
+    for arg in args:
+        print('arg {number=%d}{call=%s}{display=%s}{tooltip=%s}{type=%s}%s' % arg)
+
+    values = values + [(Config.CHANNEL.value, '%d' % i, '%d' % i, 'true' if i == 11 else 'false') for i in range(11, 27)]
+
+    for value in values:
+        print('value {arg=%d}{value=%s}{display=%s}{default=%s}' % value)
+
+def extcap_dlts(interface):
+    """List DLTs for the given interface"""
+    print('dlt {number=195}{name=IEEE802_15_4_WITHFCS}{display=IEEE 802.15.4 with FCS}')
+    print('dlt {number=283}{name=IEEE802_15_4_TAP}{display=IEEE 802.15.4 TAP}')
+
+def serialopen(interface,  __console__, DirtylogFile):
+    """
+    Open serial to indentify OpenThread sniffer
+    :param interface: string, eg: '/dev/ttyUSB0 - Zolertia Firefly platform', '/dev/ttyACM1 - nRF52840 OpenThread Device'
+    """
+    sys.stdout = DirtylogFile
+    sys.stderr = DirtylogFile
+    interface = str(interface).split()[0]
+    baudrate = None
+    result = None
+
+    for speed in COMMON_BAUDRATE:
+        stream = StreamOpen('u', interface, False, baudrate=speed)
+
+        wpan_api = WpanApi(stream, nodeid=DEFAULT_NODEID)
+        # result is None for NCP, not None for RCP
+        result = wpan_api.prop_set_value(SPINEL.PROP_PHY_ENABLED, 1) # detect baudrate
+
+        # result should not be None for both NCP and RCP
+        result = wpan_api.prop_get_value(SPINEL.PROP_CAPS) # confirm OpenThread Sniffer
+
+        # check whether or not is OpenThread Sniffer
+        stream.close()
+        if result is None:
+            continue
+        else:
+            baudrate = speed
+            break
+
+    sys.stdout = __console__
```

Comment:
better to use `try: ...finally: ...` to make sure sys.stdout, sys.stderr is restored in the end